DTLOR_DP.py

```diff
@@ -107,313 +126,221 @@ def DP(hostTree, parasiteTree, phi, locus_map, D, T, L, Origin, R):
-                            #TODO: apply combine
-                            if switch_cost < SWITCHepeh:
-                                SWITCHepeh = switch_cost
-                                # if ep2 switching has the lowest cost
-                                if ep2_switch < ep1_switch:
-                                    for location in bestSwitchLocations[(ep2,vh,l2)]:
-                                        currentLoc = location[1] # Switch landing site
-                                        switchList.append(["T", (ep1, vh, l1), (ep2, \
-                                            currentLoc, l2)])
-                                # if ep1 switching has the lowest cost
-                                elif ep1_switch < ep2_switch: 
-                                    for location in bestSwitchLocations[(ep1,vh, l1)]:
-                                        currentLoc = location[1]
-                                        switchList.append(["T", (ep2, vh, l2), \
-                                            (ep1, currentLoc, l1)])
-                                # if ep1 switching has the same cost as ep2 switching
-                                else: 
-                                    for location in bestSwitchLocations[(ep2, vh, l2)]:
-                                        currentLoc = location[1]
-                                        if currentLoc != None:
-                                            switchList.append(["T", (ep1, vh, l1), (ep2, \
-                                            currentLoc, l2)])
-                                        else:
-                                            switchList.append(["T", (ep1, vh, l1), \
-                                                (ep2, currentLoc, l2)])
-                               
-                                    for location in bestSwitchLocations[(ep1,vh, l1)]:
-                                        currentLoc = location[1]
-                                        if currentLoc != None:
-                                            switchList.append(["T", (ep2, vh, l2), \
-                                            (ep1, currentLoc, l1)])
-                                        else:
-                                            switchList.append(["T", (ep1, vh, l1), \
-                                                (ep2, currentLoc, l2)])           
-                    if switchList==[]:
-                        switchList=[[Infinity]]
-
+                    def get_transfer(l1, l2):
+                        synteny_cost = delta(l_bottom, l1) + delta(l_bottom, l2)
+                        # Cost to transfer ep2
+                        ep2_cost, ep2_locations = best_switch[(ep2, eh, l2)]
+                        ep2_switch_cost = T + synteny_cost + C[(ep1, eh, l1)][0] + ep2_cost
+                        ep2_switch_events = [("T", (ep1, vh, l1), (ep2, location[1], l2)) for location in \
```

Comment:
the edges seem to be represented by their bottom vertex in the parasite and host tree, this can be a little bit confusing. I am not sure what's the best way to go about this